chia/harvester/harvester_api.py

```diff
@@ -97,6 +112,73 @@ async def new_signage_point_harvester(
 
         loop = asyncio.get_running_loop()
 
+        def blocking_lookup_v2_partial_proofs(filename: Path, plot_info: PlotInfo) -> Optional[PartialProofsData]:
+            # Uses the V2 Prover object to lookup qualities only. No full proofs generated.
+            try:
+                plot_id = plot_info.prover.get_id()
+                sp_challenge_hash = calculate_pos_challenge(
+                    plot_id,
+                    new_challenge.challenge_hash,
+                    new_challenge.sp_hash,
+                )
+                partial_proofs = plot_info.prover.get_partial_proofs_for_challenge(sp_challenge_hash)
+
+                # If no partial proofs are found, return None
+                if partial_proofs is None or len(partial_proofs) == 0:
```

Comment:
I just realized `get_partial_proofs_for_challenge()` does not return `Optional[]`. So I think you can just check for the list being empty